### ISSUE TITLE
False positive: This condition is duplicated in another if/elseif branch

### ISSUE THREAD

**Issue by simPod** [kalessil/phpinspectionsea#1700]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | 4.0.6 |
| Language level | 7.4 |

Current behaviour
--

In the first `elseif` branch it says `[EA] This condition is duplicated in another if/elseif branch.` for `$this instanceof stdClass`

```php
function (int $p) : int
{
    if ($this instanceof stdClass && $p === 1) {
        return 1;
    } elseif ($this instanceof stdClass && $p === 2) {
        return 2;
    } elseif ($p === 3) {
        return 3;
    }
    
    return 4;
}
```

Expected behaviour
--

No suggestions, in this case it is irrelevant whether it is duplicated or not.

**Comment by orklah** [comment]
Well, one could argue you could extract the condition out of the if/else structure. That way the instruction would be executed only once

**Comment by simPod** [comment]
But the condition would be duplicated anyway 🤔

**Comment by ea-inspections-team** [comment]
Thanks for reporting @simPod!
The issue should be addressed already, the detection of the duplicate conditions was dropped in July 2020.
